src/inmanta/ast/entity.py

```diff
@@ -524,8 +524,8 @@ def __repr__(self) -> str:
     def normalize(self) -> None:
         try:
             self.statements.normalize()
-        except CompilerException as e:
-            e.set_location(self.location, False)
+        except RuntimeException as e:
```

Comment:
`set_statement` doesn't exist on CompilerException and I think this issue arises after the parsing is completed ans this is what the docstring of RuntimeException says. (I haven't checked, i'm thinking this intuitively).

Reply:
I think you're right. I did a quick check and I believe we use the appropriate exceptions everywhere except for plugins definitions. That should not be relevant here though we should probably change it to use `RuntimeException` as well: #6971.

I'm adding @wouterdb for this change just in case he sees a reason to keep this at `CompilerException`.

Reply:
@wouterdb I put the CompilerException block as well